### ISSUE TITLE
Bug when claiming bonds in the case of multiple answers on reality

### ISSUE THREAD

**Issue by kongzii** [valory-xyz/trader#310]
[According to Realitio docs](https://realitio.github.io/docs/html/contract_explanation.html#answer-history-entries), only the latest history hash is stored in the question struct:

```
Empty  Bytes32 + Answer + Bond + Answerer + False -> History Hash 1
History Hash 1 + Answer + Bond + Answerer + False -> History Hash 2
History Hash 2 + Answer + Bond + Answerer + False -> History Hash 3 <- stored in Questions struct
```

If multiple answers were submitted, and we query them on subgraph, we can see it:

![Screenshot by Dropbox Capture](https://github.com/user-attachments/assets/b0288a02-7ce7-4635-881c-10d7a6dfa3fa)

All the returned history hashes are the same. [(it's this question)](https://reality.eth.limo/app/#!/question/0x79e32ae03fb27b07c89c0c568f80287c01ca2e57-0x7e6d8446fd70f8b20de6a9925135bee8298bc15cdc5a8632d17bf09307cf1972/token/XDAI)

And that's why the following code https://github.com/kongzii/trader/blob/main/packages/valory/skills/market_manager_abci/graph_tooling/requests.py#L297:

```
answers = [
            {
                "args": {
                    "answer": bytes.fromhex(answer["answer"][2:]),
                    "question_id": bytes.fromhex(answer["question"]["questionId"][2:]),
                    "history_hash": bytes.fromhex(
                        answer["question"]["historyHash"][2:]
                    ),
                    "user": Web3.to_checksum_address(answer["question"]["user"]),
                    "bond": int(answer["bondAggregate"]),
                    "timestamp": int(answer["timestamp"]),
                    "is_commitment": False,
                }
            }
            for answer in raw_answers
]
```

submits invalid hashes if there are multiple different answers and claiming of bonds fails.

**Comment by gabrielfior** [comment]
Here's what I know
-> Subgraph only serves the latest history hash (not all previous history hashes)
-> If there's a second answer, what happens?
a. Second answer is correct - market creator will try to claim bond, unsuccessful (OK, as its answer was wrong)
b. Second answer is incorrect - market creator will try to claim bond, unsucessful (NOT OK, because its answer was correct and claim should have been possible, but history hash will not match the one from the answer)
In conclusion, fetching history_hash from the subgraph is not working for this.

**Comment by kongzii** [comment]
We fixed it in our repo at https://github.com/gnosis/prediction-market-agent-tooling/pull/368

**Comment by Adamantios** [comment]
TODO: Check if this is still relevant.
